src/components/CategoriesPills.js

```diff
@@ -0,0 +1,144 @@
+import type { StyleObj } from "react-native/Libraries/StyleSheet/StyleSheetTypes";
+import Text from "./Text";
+import text from "../constants/text";
+import eventCategories from "../constants/event-categories";
+import locale from "../data/locale";
+
+import {
+  blackColor,
+  whiteColor,
+  darkBlueGreyTwoColor,
+  eucalyptusGreenColor
+} from "../constants/colors";
+
+const categoryStyleColor = (category: string) => {
+  const categoryData = eventCategories[locale][category];
+  return { color: categoryData.contrast ? blackColor : whiteColor };
+};
+
+const categoryStyleBackgroundColor = (category: string) => {
+  const categoryData = eventCategories[locale][category];
+  return { backgroundColor: categoryData.color };
+};
+
+type Props = {
+  selectedCategories: Set<string>,
+  style?: StyleObj
+};
+
+const CategoryPill = ({ name }: { name: string }) => (
+  <View style={[styles.categoryPill, categoryStyleBackgroundColor(name)]}>
+    <Text type="h3" style={[styles.categoryPillText, categoryStyleColor(name)]}>
+      {name}
+    </Text>
+  </View>
+);
+
+class CategoriesPills extends React.PureComponent<Props> {
+  static defaultProps = {
+    style: {}
+  };
+
+  scrollView: ?Object;
+
+  scrollToLastPill = () => {
+    if (!this.scrollView) {
+      return;
+    }
+
+    this.scrollView.scrollToEnd({ animated: false });
+  };
+
+  render() {
+    const { style, selectedCategories } = this.props;
+
+    return (
+      <View style={[styles.selectedCategoriesPills, style]}>
+        {selectedCategories.size === 0 ? (
+          <Text type="h3" style={styles.zeroSelected}>
+            {text.zeroSelected}
+          </Text>
+        ) : (
+          <View style={styles.scrollView}>
+            <LinearGradient
+              style={[styles.scrollShadow, styles.scrollShadowLeft]}
+              end={{ x: 0, y: 0.5 }}
+              start={{ x: 1.0, y: 0.5 }}
+              colors={["rgba(0, 0, 0, 0)", "rgba(0, 0, 0, 0.3)"]}
+            />
+            <ScrollView
+              ref={(ref: ?Object) => {
+                this.scrollView = ref;
+              }}
+              horizontal
+              showsHorizontalScrollIndicator={false}
+              onContentSizeChange={this.scrollToLastPill}
+            >
+              {Array.from(selectedCategories).map(name => (
+                <CategoryPill key={name} name={name} />
+              ))}
+            </ScrollView>
+            <LinearGradient
+              style={[styles.scrollShadow, styles.scrollShadowRight]}
+              end={{ x: 1.0, y: 0.5 }}
+              start={{ x: 0, y: 0.5 }}
+              colors={["rgba(0, 0, 0, 0)", "rgba(0, 0, 0, 0.3)"]}
+            />
+          </View>
+        )}
+      </View>
+    );
+  }
+}
+
+const styles = StyleSheet.create({
+  selectedCategoriesPills: {
+    backgroundColor: darkBlueGreyTwoColor,
+    flex: 1,
+    alignItems: "center",
+    height: 44 * PixelRatio.getFontScale(),
```

Comment:
That's actually a nice solution. Looks very clean.

I wanted to be picky and say: "How does this work when you change the font size while the app is open?".

But then I tried it. Looks like React Native just restarts the whole app in this case, even resetting the navigation. Interesting. But makes our live easier here.